src/main/scala/nl.knaw.dans.easy.s2d/dataverse/Dataset.scala

```diff
@@ -23,7 +23,7 @@ import nl.knaw.dans.lib.logging.DebugEnhancedLogging
 
 import scala.util.{ Failure, Try }
 
-class Dataset(id: String, isPersistentId: Boolean, configuration: DataverseInstanceConfig)(implicit resultOutput: PrintStream) extends HttpSupport with DebugEnhancedLogging {
+case class Dataset(id: String, isPersistentId: Boolean, configuration: DataverseInstanceConfig)(implicit resultOutput: PrintStream) extends HttpSupport with DebugEnhancedLogging {
```

Comment:
For a testable version return the case class to an ordinary class and:

```
class Dataset(id: String, isPersistentId: Boolean, configuration: DataverseInstanceConfig)(implicit resultOutput: PrintStream) extends DebugEnhancedLogging {
  protected lazy val httpSupport = new HttpSupport {
    override protected val connectionTimeout: Int = configuration.connectionTimeout
    override protected val readTimeout: Int = configuration.readTimeout
    override protected val baseUrl: URI = configuration.baseUrl
    override protected val apiToken: String = configuration.apiToken
    override protected val apiVersion: String = configuration.apiVersion
  }
```

Your test class then can become (I hope) something like 

```
class DatasetSpec extends FlatSpec with Matchers with MockFactory{

  class MockedHttpSupport extends HttpSupport {
    val apiToken: String = null
    val connectionTimeout: Int = 0
    val apiVersion: String = null
    val baseUrl: URI = null
    val readTimeout: Int = null
  }
  class MockedDataset(id: String, isPersistentId: Boolean, configuration: DataverseInstanceConfig)(implicit resultOutput: PrintStream)
    extends Dataset(id, isPersistentId, configuration)(resultOutput) {
    override lazy val httpSupport: HttpSupport = mock[HttpSupport]
  }
```